pkg/reconciler/claimbinding/api.go

```diff
@@ -190,3 +217,25 @@ func (a *APIStatusBinder) Unbind(ctx context.Context, _ resource.Claim, mg resou
 
 	return errors.Wrap(resource.IgnoreNotFound(a.client.Delete(ctx, mg)), errDeleteManaged)
 }
+
+// equal returns true if the supplied object references are considered equal. We
+// consider two otherwise matching references with different UIDs to be equal,
+// presuming that they are both references to a particular object that has been
+// deleted and recreated, e.g. due to being restored from a backup.
+//
+// TODO(negz): If we switch to a reference that only has the fields we care
+// about (GVK, namespace, and name) we can just use struct equality.
+// https://github.com/crossplane/crossplane-runtime/issues/49
+func equal(a, b *corev1.ObjectReference) bool {
```

Comment:
Do you think this could potentially be reused elsewhere? May make sense to export it and move to a more general pkg

Reply:
This seems simple enough that I'd rather do that if and when we find ourselves wanting to write it elsewhere, rather than ahead of time.